### ISSUE TITLE
add "preview mode" for interactive rendering

### ISSUE THREAD

**Issue by ingowald** [owl-project/ViSII#75]
would be useful to have on low-end devices and/or challenging scenes - reduce sample counts etc, ideally w/ max time-per-frame budget

**Comment by n8vm** [comment]
In interactive mode, a preview is already shown in a window as the image progressively renders. This can be used in combination with the optix denoiser for extra noisy scenes to “get a sense” of what the final render will look like.

The max time-per-frame idea is a bit tricky, since per-frame render time is not known until after the image is rendered. (even one ray per pixel might take longer than the specified maximum amount of time) 

I also don’t currently support rendering less than/more than one sample per pixel (that might need to happen first). I’ve been avoiding rendering less than/more than one sample per pixel, since that complicates segmentation exports, and if not done carefully, can also close the door for temporal-spatial denoising down the road. 

So in short, I’ve thought about it, but the additional complexity vs potential benefits doesn’t seem worth the effort imo, at least not in the immediate future.
